Update dcpdig for Upload Service DB refactor

sampierson opened the pull request:
in particular, the Integer Primary Keys change

sampierson committed:
WIP

rexwangcc committed:
Update the requirements.

rexwangcc committed:
1st iteration of the analysis agents and finders.

rexwangcc committed:
Merge branch 'master' into analysis

rexwangcc committed:
Merge branch 'master' into analysis

rexwangcc committed:
Refacotr existing component entities and implement the analysis workflow entity.

rexwangcc committed:
Implement analysis agent.

rexwangcc committed:
Make sure the existing component finders use absolute import; implement analysis finder.

rexwangcc committed:
Implement analysis logic for dcpdig command

rexwangcc committed:
Implement analysis logic for analyze-submission command.

rexwangcc committed:
Update the requirements.

rexwangcc committed:
Update code according to the comments.

rexwangcc commented on x:
```diff
@@ -0,0 +1,1549 @@
+Using deployment integration
```
Curious: what is this file (looks like a log to me) and are there any credentials exposed in this file?

sampierson replied:
My bad.  I removed that file.
No credentials.  That was just the output of a `dcpdig @upload` run.

sampierson committed:
dcpdig: update for Upload Service integer primary key changes

Some column names have changed, e.g. id became uuid, and id is now
an integer sequence.

Querying for file_id no longer seems useful, so let file= query for
the S3 key.

sampierson committed:
dcpdig: restructure __init__ arg processing to eliminate ugly "if"

Finders much now all take **args.
